app/validation/validator.py

```diff
@@ -151,6 +153,75 @@ def validate_range_types_from_answers(self, json_to_validate):
 
         return errors
 
+    def _validate_min_max_ranges(self, json_to_validate):
+        errors = []
+        answers = {}
+
+        for block in self._get_blocks(json_to_validate):
+            for answer in self._get_answers_for_block(block):
+                answers[answer.get('id')] = answer
+
+        for answer_id, answer in answers.items():
```

Comment:
Do you need this 2nd loop? if you remove this and nest the code below inside the previous loop it should keep working.
As any referenced answer should be before the answer that it is referenced from?